pmd-core/src/main/java/net/sourceforge/pmd/cpd/JavaCCTokenizer.java

```diff
@@ -0,0 +1,42 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.cpd;
```

Comment:
Hm... not sure. I would consider this still an API - you typically subclass this abstract class instead of implementing the whole Tokenizer interface directly, when you're implementing support for a new language.

We could mark it as \@Experimental, since it adds a couple of protected methods - which might or might not change in the future...

Reply:
I'll move this to the internal package.

Reply:
I agree with you @adangel, this is intended as a utility for new languages to subclass, and therefore probably public API... Using `@Experimental` would be an option if we feel this may change in the future, and we possibly graduate it in a future release (8.0.0?).

Marking it internal is maybe too strong a message to third party developers.